src/mds/MDSRank.cc

```diff
@@ -1498,6 +1498,11 @@ void MDSRank::boot_start(BootStep step, int r)
       ceph_assert(r == 0);  // Unreachable, damaged() calls respawn()
     } else if (r == -EROFS) {
       dout(0) << "boot error forcing transition to read-only; MDS will try to continue" << dendl;
+    } else if (r == -EBLACKLISTED) {
+      // we have been blacklisted and we respawned too early
+      dout(0) << "boot_start encountered EBLACKLISTED error, failing" << dendl;
+      suicide();
+      return;
```

Comment:
Below, `onfinish->complete` should be calling `MDSIOContextBase::complete`. That already has a blacklisted check. Did you encounter a problem that prompted this change? I would think only the change in `src/osdc/Journaler.cc` is necessary.

Reply:
oh! I hadn't investigated the `MDSIOContextBase::complete` implementation
The `MDSIOContextBase::complete` implementation respawns MDS when EBLACKLISTED is encountered. Wouldn't this result in a respawn loop, since the MDS (standby-replay) has already been blacklisted.

Reply:
No, the MDS gets a new instance ID and resumes becoming standby.